lively.ide/world.js

```diff
@@ -1535,3 +1541,36 @@ export class LivelyWorld extends World {
     });
   }
 }
+
+class FileWatcher {
+  constructor () {
+    this.fileActions = {};
+    this.onFileSave.bind(this);
+    subscribe('file/save', this.onFileSave.bind(this));
+  }
+
+  clear () {
+    this.fileActions = {};
+  }
+
+  registerFileAction (file, cB) {
+    if (this.fileActions[file]) {
+      if (Array.isArray(this.fileActions[file])) this.fileActions[file].push(cB);
+      else this.fileActions[file] = [this.fileActions[file], cB];
+    } else { this.fileActions[file] = cB; }
+  }
+
+  unregisterFileActions (file) {
+    this.fileActions[file] = null;
+  }
+
+  async onFileSave (savedFileData) {
+    if (!this.fileActions[savedFileData.resource]) return;
```

Comment:
Can we also make this match regular expressions instead of the literal file names? That would allow us to perform general action on when a css file changes, such as updating it in all browsers.

Reply:
Yes, we can certainly do that. I would however prefer to do that separately and not as part of this PR, is that okay with you?